File: app/src/main/java/io/github/gmazzo/android/livewallpaper/weather/engine/GlobalRand.java
package io.github.gmazzo.android.livewallpaper.weather.engine;

import java.util.Random;

public class GlobalRand {
    public static Random rand = new Random();

    public static boolean flipCoin() {
        return rand.nextFloat() < 0.5f;
    }

    public static float floatRange(float min, float max) {
        return (rand.nextFloat() * (max - min)) + min;
    }

    public static int intRange(int min, int max) {
        return rand.nextInt(max - min) + min;
    }

    public static void randomNormalizedVector(EngineColor dest) {
        dest.set(new Vector(
                        floatRange(-1.0f, 1.0f),
                        floatRange(-1.0f, 1.0f),
                        floatRange(-1.0f, 1.0f))
                        .normalize(),
                dest.getA());
    }

}

File: app/src/main/kotlin/io/github/gmazzo/android/livewallpaper/weather/engine/GlobalRand.kt
package io.github.gmazzo.android.livewallpaper.weather.engine

import java.util.Random

object GlobalRand {
    var rand: Random = Random()

    fun flipCoin(): Boolean {
        return rand.nextFloat() < 0.5f
    }

    fun floatRange(min: Float, max: Float): Float {
        return (rand.nextFloat() * (max - min)) + min
    }

    fun intRange(min: Int, max: Int): Int {
        return rand.nextInt(max - min) + min
    }

    fun randomNormalizedVector(dest: EngineColor) {
        dest.set(
            Vector(
                floatRange(-1.0f, 1.0f),
                floatRange(-1.0f, 1.0f),
                floatRange(-1.0f, 1.0f)
            )
                .normalize(),
            dest.a
        )
    }
}
